### DISCUSSION TITLE
Missing doc for history property on router and RouterProvider in API Components

### DISCUSSION THREAD

**Discussion by GYHHAHA** [TanStack/router#1583] [General]
I didn't have any experience with React Router before learning TanStack Router, and when I found `router.history.push(search.redirect)` from [here](https://tanstack.com/router/latest/docs/framework/react/guide/authenticated-routes#redirecting), it was confusing for:
- I didn't find there is a history property which can be access from [Router Class](https://tanstack.com/router/latest/docs/framework/react/api/router/RouterClass) (but available from [RouterOptions](https://tanstack.com/router/latest/docs/framework/react/api/router/RouterOptionsType))
- I hope there is a section to show what `history.push` do, which might be useful for those who lack of `React Router` exp, and its difference with `history.replace` (if there is a `replace` method available from the `history` object because I also didn't find there was a doc to describe the `history` structure)

Another point is `RouterProvider` is not showed in API Components section, but it's widely used in the doc guides. Since many util-components like `Outlet` and `Await` are documented in API, I wondering is there a specific reason to omit `RouterProvider`?

**Comment by SeanCassiere** [comment]
The `<RouterProvider />` accepts all the same options as [RouterOptions](https://tanstack.com/router/latest/docs/framework/react/api/router/RouterOptionsType) + the [Router](https://tanstack.com/router/latest/docs/framework/react/api/router/createRouterFunction#createrouter-returns) instance.

These are the type for [RouterHistory](https://github.com/TanStack/router/blob/f6e9ab3b60ca42401923648649930db4ae97fc00/packages/history/src/index.ts#L5). TSR's history package, uses pretty much all of the same browser standards stuff with some optimizations specific to the library.
